php/src/Docker/DockerActionManager.php

```diff
@@ -35,98 +38,57 @@ private function BuildImageName(Container $container) : string {
-
-        if ($responseBody['State']['Restarting'] === true) {
-            return ContainerState::Restarting;
+        $body = json_decode($response->getBody()->getContents(), true);
+        assert(is_array($body));
+        assert(is_array($body['State']));
+
+        $state = match ($body['State']['Status']) {
+            'running' => ContainerState::Running,
+            'created' => ContainerState::Starting,
+            'restarting' => ContainerState::Restarting,
+            'paused', 'removing', 'exited', 'dead' => ContainerState::Stopped,
+            default => throw new AssertionError()
+        };
+
+        if ($state === ContainerState::Running && is_array($body['State']['Health'])) {
+            return match ($body['State']['Health']['Status']) {
+                'starting' => ContainerState::Starting,
+                'unhealthy' => ContainerState::Unhealthy,
+                default => ContainerState::Running,
+            };
         } else {
-            return ContainerState::NotRestarting;
+            return $state;
         }
     }
 
-    public function GetContainerUpdateState(Container $container) : VersionState
-    {
+    public function GetUpdateState(Container $container): UpdateState {
         $tag = $container->GetImageTag();
         if ($tag === '%AIO_CHANNEL%') {
             $tag = $this->GetCurrentChannel();
         }
 
         $runningDigests = $this->GetRepoDigestsOfContainer($container->GetIdentifier());
         if ($runningDigests === null) {
-            return VersionState::Different;
+            return UpdateState::Outdated;
         }
         $remoteDigest = $this->dockerHubManager->GetLatestDigestOfTag($container->GetContainerName(), $tag);
         if ($remoteDigest === null) {
-            return VersionState::Equal;
-        }
-
-        foreach($runningDigests as $runningDigest) {
-            if ($runningDigest === $remoteDigest) {
-                return VersionState::Equal;
-            }
-        }
-        return VersionState::Different;
-    }
-
-    public function GetContainerStartingState(Container $container) : ContainerState
-    {
-        $runningState = $this->GetContainerRunningState($container);
-        if ($runningState === ContainerState::Stopped || $runningState === ContainerState::ImageDoesNotExist) {
-            return $runningState;
-        }
-
-        $containerName = $container->GetIdentifier();
-        $internalPort = $container->GetInternalPort();
-        if($internalPort === '%APACHE_PORT%') {
-            $internalPort = $this->configurationManager->GetApachePort();
-        } elseif($internalPort === '%TALK_PORT%') {
-            $internalPort = $this->configurationManager->GetTalkPort();
+            return UpdateState::Latest;
         }
 
-        if ($internalPort !== "" && $internalPort !== 'host') {
-            $connection = @fsockopen($containerName, (int)$internalPort, $errno, $errstr, 0.2);
```

Comment:
@docjyJ we need to keep this. (I oversaw this)

Reply:
I put the connection test back. A container marked healthy but not reachable will be in the starting state